### PR TITLE
[onert] Refactoring MoelTestDynamicTensor.cc

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
This refactors `MoelTestDynamicTensor.cc`:
1. Rename `set_input_output` to `setInputOutput` (came case)
2. Make `setInputOutput()` accept template inputs
3. Create `verifyOutput()` func

Originated from draft #4627
Excerpted from #4718

Signed-off-by: Hyun Sik Yoon <hyunsik.yoon.1024@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/ModelTestDynamicTensor.cc [hunk 1]
```diff
@@ -22,31 +22,89 @@
...
+// if there's any error while setting input or output:
+//
+//  EXPECT_ANY_THROW(setInputOutput(...));
+//
+#define THROW_WHEN_NNFW_ERROR(result)                                  \
+  do                                                                   \
+  {                                                                    \
+    if (result != NNFW_STATUS_NO_ERROR)                                \
+      throw std::runtime_error("returning error on calling nnfw api"); \
+  } while (false);
```
Comment (wateret) [inline tests/nnfw_api/src/ModelTestDynamicTensor.cc:36]: I would like to suggest 

```suggestion
  } while (false)
```

I think it is more common used without `;`. (to force developers always put `;` when they use this macro)

**Comment by wateret** [comment]
👍 I think some functions can be reused from other tests, later.

File: tests/nnfw_api/src/ModelTestDynamicTensor.cc (hunk 1 above)
Comment (hyunsik-yoon) [reply tests/nnfw_api/src/ModelTestDynamicTensor.cc:36]: super!

**Comment by hyunsik-yoon** [comment]
PTAL

**Review by wateret — APPROVED** [review]
LGTM

**Pull request merged**